lib/sdr_client/deposit/files/direct_upload_request.rb

```diff
@@ -10,19 +10,36 @@ def self.from_file(path, file_name:, content_type:)
           checksum = Digest::MD5.file(path).base64digest
           new(checksum: checksum,
               byte_size: ::File.size(path),
-              content_type: content_type || 'application/octet-stream',
+              content_type: clean_content_type(content_type),
               filename: file_name)
         end
 
         def as_json
           {
-            blob: { filename: filename, byte_size: byte_size, checksum: checksum, content_type: content_type }
+            blob: { filename: filename, byte_size: byte_size, checksum: checksum,
+                    content_type: self.class.clean_content_type(content_type) }
           }
         end
 
         def to_json(*_args)
           JSON.generate(as_json)
         end
+
+        def self.clean_content_type(content_type)
```

Comment:
should this be a case statement on content-type and return it as is unless it's one of the three cases below?  Or is this only called for json?

Reply:
i can change to a case statement if it looks clearer, logic will be the same